SimulationAgent/SimulationManager.cs

```diff
@@ -466,6 +470,13 @@ private async Task<bool> TryToCreateActorsForPartitionAsync(DevicesPartition par
                     this.CreateActorsForDevice(deviceId, deviceModel, this.deviceCount);
                     this.deviceCount++;
                     count++;
+
+                    // Set ActualStartTime if required
+                    if (!this.simulation.ActualStartTime.Value.Equals(DateTimeOffset.MinValue))
```

Comment:
since this is nullable (see also my other comment) you can simplify this expression using `null` and `HasValue`

Reply:
updated